Runtimes/Supplemental/Synchronization/cmake/caches/Vendors/Apple/Darwin.toolchain.cmake

```diff
@@ -0,0 +1,106 @@
+# Toolchain configuration when compiling for Darwin Platforms
```

Comment:
We can likely share the toolchain file since it's only setting up the compilers and such that are used while compiling.
I don't think that this is setting anything specific to the Synchronization library. It's just the cache files that are setting project-specific options.

Maybe stick it in `Runtimes/Supplemental/cmake/toolchains/Vendors/Apple/Darwin.toolchain.cmake`?

@compnerd, do you have any opinions on the file layout for the toolchain files?

Reply:
I suppose I am left wondering why are the caches in `Runtimes/Supplemental/cmake/toolchains/Vendors/Apple` rather than say `Runtimes/cmake/caches/` and `Runtimes/cmake/toolchains/`.

Reply:
I think keeping the project-specific cache files in a subdirectory of the project itself makes more sense. The `PROJECT_NAME` isn't set until the `project` call, so that would get kind of awkward.

As for the toolchain file, I think it's fine to put that in `Runtimes/cmake/toolchains/...`. For syncing things on our side, we won't be able to add it there and remove it from core in the same PR, but it shouldn't need to be in there for too long.